crates/router/src/services/authentication.rs

```diff
@@ -1113,6 +1113,91 @@ where
     }
 }
 
+pub struct JWTAuthMerchantAndProfileFromRoute {
+    pub merchant_id: id_type::MerchantId,
+    pub profile_id: String,
+    pub required_permission: Permission,
+}
+
+impl JWTAuthMerchantAndProfileFromRoute {
+    pub fn new(
+        merchant_id_from_route: id_type::MerchantId,
+        permission: Permission,
+        profile_id_from_route: String,
+    ) -> Self {
+        Self {
+            merchant_id: merchant_id_from_route,
+            required_permission: permission,
+            profile_id: profile_id_from_route,
+        }
+    }
+}
```

Comment:
NIT: We generally don't use constructors for auth types. For consistency this can be removed. Not a required change though.